python/ray/_private/runtime_env/packaging.py

```diff
@@ -768,7 +883,12 @@ def delete_package(pkg_uri: str, base_directory: str) -> Tuple[bool, int]:
     """
 
     deleted = False
-    path = Path(_get_local_path(base_directory, pkg_uri))
+    local_path, uri_type = _get_local_path(base_directory, pkg_uri)
+    if uri_type is not UriType.REMOTE:
+        raise RuntimeError(
+            f"The deletion of the package of {uri_type} type is forbidden."
+        )
```

Comment:
The name UriType.REMOTE is a little confusing, since we're deleting from a local filesystem, but I don't know a better name offhand--I'll let you know if I think of one.